### PR TITLE
Introduce logging level configuration

### PR DESCRIPTION
**Pull request by pavolloffay**
Logging struct was defined in flag package but not used. I have also removed unused code from there.

### CODE REVIEW

File: cmd/collector/main.go [hunk 1]
```diff
@@ -126,6 +134,7 @@ func main() {
 			case <-signalsChannel:
 				logger.Info("Jaeger Collector is finishing")
 			}
+			select {}
```
Comment (yurishkuro) [inline cmd/collector/main.go]: why is this needed?

File: cmd/flags/flags.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ const (
 	// ESStorageType is the storage type flag denoting an ElasticSearch backing store
 	ESStorageType                  = "elasticsearch"
 	spanStorageType                = "span-storage.type"
-	logLevel                       = "log-level"
+	logLevel                       = "log-Level"
```
Comment (yurishkuro) [inline cmd/flags/flags.go]: shouldn't it be lower case?

File: cmd/query/main.go [hunk 3]
```diff
@@ -118,6 +126,7 @@ func main() {
 			case <-serverChannel:
 				logger.Info("Jaeger Query is finishing")
 			}
+			select {}
```
Comment (yurishkuro) [inline cmd/query/main.go]: ?

File: cmd/flags/flags.go [hunk 4]
```diff
@@ -60,19 +72,22 @@ type SharedFlags struct {
...
 	flags.DependencyStorage.DataFrequency = v.GetDuration(dependencyStorageDataFrequency)
+	flags.Logging.Level = v.GetString(logLevel)
 	return flags
 }
 
 // AddFlags adds flags for SharedFlags
 func AddFlags(flagSet *flag.FlagSet) {
 	flagSet.String(spanStorageType, CassandraStorageType, fmt.Sprintf("The type of span storage backend to use, options are currently [%v,%v,%v]", CassandraStorageType, ESStorageType, MemoryStorageType))
-	flagSet.String(logLevel, "info", "Minimal allowed log level")
+	flagSet.String(logLevel, "info", "Minimal allowed log Level. For more levels see zap logger documentation")
```
Comment (yurishkuro) [inline cmd/flags/flags.go]: instead of `zap logger documentation` I'd print the url

File: cmd/flags/flags.go [hunk 5]
```diff
@@ -44,14 +44,26 @@ func AddConfigFileFlag(flagSet *flag.FlagSet) {
...
 		if err != nil {
-			logger.Fatal("Error loading config file", zap.Error(err), zap.String(configFile, file))
+			errors.Wrap(err, fmt.Sprintf("Error loading config file %s", file))
 		}
 	}
+	return nil
+}
+
+// NewLogger pareses log Level
+func (flags *SharedFlags) NewLogger(conf zap.Config, options ...zap.Option) (*zap.Logger, error) {
```
Comment (yurishkuro) [inline cmd/flags/flags.go]: it looks like in all cases you're passing zap.Production() as the config. Is there a use case you're thinking of when it's not going to be production?  Also, do we need a way to affect that config from cmd line / our config?

File: cmd/flags/flags.go [hunk 6]
```diff
@@ -44,14 +44,26 @@ func AddConfigFileFlag(flagSet *flag.FlagSet) {
...
 		err := v.ReadInConfig()
 		if err != nil {
-			logger.Fatal("Error loading config file", zap.Error(err), zap.String(configFile, file))
+			errors.Wrap(err, fmt.Sprintf("Error loading config file %s", file))
 		}
 	}
+	return nil
+}
+
+// NewLogger pareses log Level
```
Comment (yurishkuro) [inline cmd/flags/flags.go]: typo "pareses"

File: cmd/flags/flags.go [hunk 7]
```diff
@@ -88,19 +103,5 @@ type spanStorage struct {
...
-	MaxRetryAttempts   int
-	QueryTimeout       time.Duration
-	Servers            string
-	Port               int
-	Keyspace           string
-}
-
-func (co cassandraOptions) ServerList() []string {
-	return strings.Split(co.Servers, ",")
-}
```
Comment (yurishkuro) [inline cmd/flags/flags.go]: nice catch

File: cmd/query/main.go [hunk 8]
```diff
@@ -132,7 +141,8 @@ func main() {
 	)
 
 	if error := command.Execute(); error != nil {
-		logger.Fatal(error.Error())
+		fmt.Println(error.Error())
```
Comment (yurishkuro) [inline cmd/query/main.go]: this is unfortunate, because a logger may print the error quite differently, e.g. with a stack trace, write to file, report to Sentry, etc. If we guaranteed that the command only returned errors when it can't init the logger, then it wouldn't be too bad, but I don't think this is generally the case.

File: cmd/flags/flags.go (hunk 4 above)
Comment (pavolloffay) [reply cmd/flags/flags.go]: added but it looks a bit weird. 

Message for conf file: 
```
Configuration file in JSON, TOML, YAML, HCL, or Java properties formats (default none). See spf13/viper for precedence.
```

File: cmd/flags/flags.go (hunk 5 above)
Comment (pavolloffay) [reply cmd/flags/flags.go]: > Is there a use case you're thinking of when it's not going to be production? Also, do we need a way to affect that config from cmd line / our config?

Not right now. Maybe a debug flag or something like that in the future. I made this as flexible as possible. 

So do you want to change it to the following?
```go
func (flags *SharedFlags) NewLogger() (*zap.Logger, error) 
```

File: cmd/query/main.go (hunk 8 above)
Comment (pavolloffay) [reply cmd/query/main.go]: This statement is invoked when: 
* an error is propagated from `runE` -> in all cases except logger creation and config file parsing `zap.Fatal` is called which directly calls `os.exit`.
* cobra fails to parse flags e.g. a wrong/undefined flag

Other solution might be to use a default logger here.

File: cmd/flags/flags.go [hunk 9]
```diff
@@ -44,14 +44,26 @@ func AddConfigFileFlag(flagSet *flag.FlagSet) {
...
 
 // TryLoadConfigFile initializes viper with config file specified as flag
-func TryLoadConfigFile(v *viper.Viper, logger *zap.Logger) {
+func TryLoadConfigFile(v *viper.Viper) error {
 	if file := v.GetString(configFile); file != "" {
 		v.SetConfigFile(file)
 		err := v.ReadInConfig()
 		if err != nil {
-			logger.Fatal("Error loading config file", zap.Error(err), zap.String(configFile, file))
+			errors.Wrap(err, fmt.Sprintf("Error loading config file %s", file))
```
Comment (yurishkuro) [inline cmd/flags/flags.go]: missing `return`?  And you can use `Wrapf` for extra formatting.

Comment (pavolloffay) [reply cmd/flags/flags.go]: :+1: ouch, the change from log Fatal :)

**Comment by pavolloffay** [comment]
@yurishkuro could you please re-review?

**Comment by yurishkuro** [comment]
@pavolloffay could you please rebase from master to make sure all is clean? Then ok to merge.

**Comment by pavolloffay** [comment]
@yurishkuro thanks, rebased, I will merge on green
